### ISSUE TITLE
Prune large files from git history

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#3124]
The repo is currently about 300MB to clone, the vast majority of which is the multiple versions of various JARs that we bundle and which we used to version in the repo, the biggest offender being Jython.

Here's a list of the files bigger than 5MB which adds up to over 230MB:

```
5fcc65a1d012  5.4MiB main/webapp/WEB-INF/lib/poi-ooxml-schemas-3.12-20150511.jar
bd883978b3ed  5.4MiB main/webapp/WEB-INF/lib/poi-ooxml-schemas-3.13-20150929.jar
bf0d532cb19e  6.0MiB main/webapp/WEB-INF/lib/icu4j-4.2.1.jar
56b9ce471741  6.4MiB extensions/jython/module/MOD-INF/lib/jython-2.5.1.jar
82577ca66f3a  7.9MiB extensions/jython/module/MOD-INF/lib/jython-2.5.3.jar
66d6cc8b4c68  9.1MiB lib-src/jackson-1.5.1-sources.zip
fe76618a55ff   12MiB main/webapp/WEB-INF/lib-src/jackson-src-1.9.5.zip
2203d5e19d8e   12MiB main/webapp/WEB-INF/lib-src/jackson-src-1.9.9.zip
9e01e5caa0c3   14MiB extensions/jython/module/MOD-INF/lib/jython-standalone-2.5.3.jar
5d511d3288a7   14MiB main/tests/data/ozone_8hr_dmax.csv
a0ca6c14e6bb   14MiB tests/data/ozone_8hr_dmax.csv
97c17e63ce22   15MiB main/webapp/WEB-INF/lib-src/icu4j-4.2.1-sources.jar
c49498347a6a   35MiB extensions/jython/module/MOD-INF/lib/jython-standalone-2.7.0.jar
85386dd3b3da   38MiB main/webapp/WEB-INF/lib-src/poi-3.7-sources.jar
d351edd45b3f   39MiB extensions/jython/module/MOD-INF/lib/jython-standalone-2.7.1.jar
```

`main/tests/data/ozone_8hr_dmax.csv` should be gzipped, subsetted, or deleted, and the rest of the files should be deleted from the history. They are all readily findable if we were ever to need to recreate an ancient version (unlikely).

A tool which can be used for this is https://rtyley.github.io/bfg-repo-cleaner/

**Comment by thadguidry** [comment]
@tfmorris yeap!  bfg works great!

**Comment by wetneb** [comment]
I agree it would be nice in principle but this will change all the commit ids and render all releases before the Maven migration unusable. People might have trouble updating their forks and this is likely to disrupt PRs. This should best be done when the repository is very quiet.

**Comment by thadguidry** [comment]
@wetneb so maybe after GSoC and Outreachy this year?  What's a good date to plan for on that?  After Sept 30?

**Comment by wetneb** [comment]
To be honest it is not clear to me that the ant-based releases are old enough for us to afford breaking them in the versioning system.
Making sure the software runs at all tags (and ideally all commits) is generally a good idea, for instance to preserve the ability to bisect to locate a bug.

**Comment by thadguidry** [comment]
@wetneb You have the ability to bisect for the next 10,000 years in quartz glass platters. https://archiveprogram.github.com/
Just kinda joking here of course.

But more serious, in the Enterprise we typically **would** handle this case as an archive problem.
Even though we could archive our current repo to read-only for bisect use case, adn then create a new repo (with all the CI/CD, config, etc) is a lot of work.  So we can't do it that way.  Instead, its better to flip it around, prune the current so its lightweight for active development again.  Archive the current into a new archive repo for bisect needs in the future or whatever.  Regardless, we have our repo available for the next 10,000 years it seems in the worst case :-)

So pruning the existing repo is a valid option.
I don't see a need for ensuring older tags are run-able in this current dev repo.  That need could be done during the prune process and creating a read-only archive in a new archive repo in GitHub if you want to keep history warm for older tags.  Tom has experience on handling that as he did the move for us before from Google Code to GitHub.

**Comment by tfmorris** [comment]
I should have mentioned the changed commit hashes (and meant to). 300 MB isn't completely unmanageable, so this isn't critical, but it's an extra burden for new contributors on slow internet connections.

However, as for 

> render all releases before the Maven migration unusable

that's certainly not true. All previous releases will continue to run exactly as they have. What **will** take extra work is patching old releases. Building a 2.6.1, for example, would require acquiring all of the necessary JARs. Of course, one convenient source for them would be the binary kit for the release that's being patched. We could also use Thad's idea of making a backup archive repo.

> People might have trouble updating their forks and this is likely to disrupt PRs. This should best be done when the repository is very quiet.

Yes, we would definitely want to give developers plenty of notice so that they could minimize any pending work in progress.

I'm not wedded to the idea, but I think it would make the repo easier to work with for folks.
